src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1078,7 +1078,7 @@ async def query(self, filters, options):
 
     @private
     async def legacy_afp_check(self, data, schema, verrors):
-        to_check = Path(data['path']).resolve(strict=True)
+        to_check = Path(data['path']).resolve(strict=False)
```

Comment:
Is this to keep backwards compatibility with those cases where the path/dataset might not exist anymore but we still have it in database ?

Reply:
I wish.... there's ancient SMB plugin behavior that I forgot about where we create the share path if it doesn't exist. I'm probably going to make this some sort of non-default option once we get BETA out the door.